app/models/purchase.rb

```diff
@@ -1661,9 +1661,20 @@ def cancel_setup_intent!
 
   def set_price_and_rate
     if offer_code.present? && !has_cached_offer_code?
-      self.build_purchase_offer_code_discount(offer_code:, offer_code_amount: offer_code.amount, offer_code_is_percent: offer_code.is_percent?,
-                                              pre_discount_minimum_price_cents: minimum_paid_price_cents_per_unit_before_discount,
-                                              duration_in_months: link.is_tiered_membership? ? offer_code.duration_in_months : nil)
+      if is_installment_payment && subscription&.last_payment_option&.installment_plan_snapshot&.has_original_offer_code?
+        snapshot = subscription.last_payment_option.installment_plan_snapshot
+        self.build_purchase_offer_code_discount(
+          offer_code:,
```

Comment:
In case of an installments-based purchase, in the event the offer code has been deleted in between, this will raise an error because the `offer_code` is required